app/src/main/java/de/danoeh/antennapod/fragment/preferences/StoragePreferencesFragment.java

```diff
@@ -129,45 +144,82 @@ private void setupStorageScreen() {
                             }
-                        }
-                        context.startActivity(Intent.createChooser(sendIntent,
-                                context.getResources().getText(R.string.send_label)));
-                    });
-                    alert.create().show();
-                }, error -> {
-                    alert.setTitle(R.string.export_error_label);
-                    alert.setMessage(error.getMessage());
-                    alert.show();
-                }, progressDialog::dismiss);
+                            context.startActivity(Intent.createChooser(sendIntent,
+                                    context.getResources().getText(R.string.send_label)));
+                        });
+                        alert.create().show();
+                    }, error -> {
+                        alert.setTitle(R.string.export_error_label);
+                        alert.setMessage(error.getMessage());
+                        alert.show();
+                    }, progressDialog::dismiss);
+        } else {
+            Observable<DocumentFile> observable = new DocumentFileExportWorker(exportWriter, context, uri).exportObservable();
+            disposable = observable.subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(output -> {
+                        alert.setTitle(R.string.export_success_title);
+                        String message = context.getString(R.string.export_success_sum, output.getUri());
+                        alert.setMessage(message);
+                        alert.setPositiveButton(R.string.send_label, (dialog, which) -> {
+                            Intent sendIntent = new Intent(Intent.ACTION_SEND);
+                            sendIntent.putExtra(Intent.EXTRA_SUBJECT,
+                                    context.getResources().getText(R.string.opml_export_label));
+                            sendIntent.putExtra(Intent.EXTRA_STREAM, output.getUri());
+                            sendIntent.setType("text/plain");
+                            sendIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                                List<ResolveInfo> resInfoList = context.getPackageManager().queryIntentActivities(sendIntent, PackageManager.MATCH_DEFAULT_ONLY);
+                                for (ResolveInfo resolveInfo : resInfoList) {
+                                    String packageName = resolveInfo.activityInfo.packageName;
+                                    context.grantUriPermission(packageName, output.getUri(), Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                                }
+                            }
+                            context.startActivity(Intent.createChooser(sendIntent,
+                                    context.getResources().getText(R.string.send_label)));
```

Comment:
There is still some duplicate code when creating the dialog. Please create a new function that gets the uri and then  creates and displays the dialog. The error dialog could be a second function that gets the error and then creates and displays the dialog.

Reply:
Fixed!